src/main/java/pl/psnc/dei/ui/components/SearchResultsComponent.java

```diff
@@ -280,6 +285,15 @@ public int setRowsPerPage(int rowsPerPage) {
         } else {
             rowsCount.setValue(this.rowsPerPage);
         }
+        setOldRowsLimit(this.rowsPerPage);
         return this.rowsPerPage;
     }
+
+    private int getOldRowsLimit() {
+        return UI.getCurrent().getSession().getAttribute("oldPageLimit") != null? (int) UI.getCurrent().getSession().getAttribute("oldPageLimit") : 0;
+    }
+
+    private void setOldRowsLimit(int oldPageCount) {
+        UI.getCurrent().getSession().setAttribute("oldPageLimit", oldPageCount);
```

Comment:
To be honest I do not understand why you called this attribute oldPageLimit...I know that it is just the name but it's misleading...maybe we should call it e.g. userRowsPerPage or something like this? If you change it also other naming related to this should be changed...